### PR TITLE
redis: documentation

### PR DESCRIPTION
**Pull request by danielhochman**

### CODE REVIEW

File: docs/configuration/network_filters/redis_proxy_filter.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    "config": {
+      "cluster_name": "...",
+      "conn_pool": "{...}",
+      "stat_prefix": "..."
+    }
+  }
+
+cluster_name
+  *(required, string)* Name of cluster from cluster manager.
+  See the :ref:`configuration section <arch_overview_redis_configuration>` of the architecture overview for recommendations
```
Comment (mattklein123) [inline docs/configuration/network_filters/redis_proxy_filter.rst]: nit: please try to break all around 100 cols

File: docs/configuration/network_filters/redis_proxy_filter.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  on configuring the backing cluster.
```
Comment (mattklein123) [inline docs/configuration/network_filters/redis_proxy_filter.rst]: Per offline convo we should probably do a PR to just block cluster config that does not configure ring hash, or at least warn.

File: docs/configuration/network_filters/redis_proxy_filter.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+-----------------------------
+
+.. code-block:: json
+
+  {
+    "op_timeout_ms": "...",
+  }
+
+op_timeout_ms
+  *(required, integer)* Per-operation timeout in milliseconds.
```
Comment (mattklein123) [inline docs/configuration/network_filters/redis_proxy_filter.rst]: Can you add a bit of text on how the timeout actually works. If you describe this in the arch overview feel free to link to that.

File: docs/intro/arch_overview/redis.rst [hunk 4]
```diff
@@ -0,0 +1,190 @@
+.. _arch_overview_redis:
+
+Redis
+=======
+
+Envoy can act as a Redis proxy, partitioning commands among instances in a cluster.
+In this mode we designed for availability and partition tolerance over consistency.
```
Comment (mattklein123) [inline docs/intro/arch_overview/redis.rst]: nit: rephrase without "we"

File: docs/intro/arch_overview/redis.rst [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+This is the key point when comparing Envoy to Redis Cluster.
```
Comment (mattklein123) [inline docs/intro/arch_overview/redis.rst]: link to Redis Cluster

File: docs/intro/arch_overview/redis.rst [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+-------------
+
+For filter configuration details, see the Redis proxy filter :ref:`configuration reference <config_network_filters_redis_proxy>`.
+
+The corresponding cluster definition should be configured with :ref:`ring hash load balancing <config_cluster_manager_cluster_lb_type>`.
+
+If active healthchecking is desired, the cluster should be configured with a :ref:`Redis healthcheck <config_cluster_manager_cluster_hc>`.
+
+If passive healthchecking is desired, also configure :ref:`outlier detection <config_cluster_manager_cluster_outlier_detection_summary>`.
+For the purposes of passive healthchecking, connect timeouts, command timeouts, and connection close map to 5xx. All other responses from
```
Comment (mattklein123) [inline docs/intro/arch_overview/redis.rst]: nit: line break

File: docs/configuration/network_filters/redis_proxy_filter.rst [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+Redis proxy
+===========
+
+Redis :ref:`architecture overview <arch_overview_redis>`.
+
+.. code-block:: json
+
+  {
+    "name": "redis_proxy",
+    "type": "read",
```
Comment (rshriram) [inline docs/configuration/network_filters/redis_proxy_filter.rst]: nit..we no longer use types right?

**Pull request merged**